sdk/core/src/CommonNetworkMember.ts

```diff
@@ -95,30 +96,50 @@ import {
   PROD_METRICS_URL,
 } from './_defaultConfig'
 
+type Constructor<T> = new (
+  password: string,
+  encryptedSeed: string,
+  options?: SdkOptions,
+  component?: EventComponent,
+) => T
+type AbstractStaticMethods<T> = {
+  afterConfirmSignUp: (networkMember: T, originalOptions: SdkOptions) => Promise<void>
+}
+type ConstructorKeys<T> = {
+  [P in keyof T]: T[P] extends new (...args: unknown[]) => unknown ? P : never
+}[keyof T]
+type OmitConstructor<T> = Omit<T, ConstructorKeys<T>>
+type DerivedThis<T> = Constructor<T> & AbstractStaticMethods<T> & OmitConstructor<typeof CommonNetworkMember>
+
 @profile()
-export class CommonNetworkMember {
+export abstract class CommonNetworkMember {
   private readonly _api: API
   private _did: string
-  private _apiKey: string
-  private _accessApiKey: string
+  private readonly _accessApiKey: string
   private readonly _encryptedSeed: string
   private _password: string
-  private readonly _walletStorageService: WalletStorageService
+  protected readonly _walletStorageService: WalletStorageService
   private readonly _revocationService: RevocationService
-  private readonly _keysService: KeysService
+  protected readonly _keysService: KeysService
   private readonly _jwtService: JwtService
   private readonly _holderService: HolderService
   private readonly _metricsService: MetricsService
   private readonly _didDocumentService: DidDocumentService
-  private _affinity: Affinity
-  private readonly _sdkOptions: SdkOptionsWithCongitoSetup
+  private readonly _affinity: Affinity
+  protected readonly _sdkOptions: SdkOptionsWithCongitoSetup
   private readonly _phoneIssuer: PhoneIssuerService
   private readonly _emailIssuer: EmailIssuerService
   private _didDocumentKeyId: string
-  protected _component: EventComponent
+  protected readonly _component: EventComponent
   protected cognitoUserTokens: CognitoUserTokens
 
-  constructor(password: string, encryptedSeed: string, options: SdkOptions = {}, component?: EventComponent) {
+  constructor(
+    password: string,
+    encryptedSeed: string,
+    platformEncryptionTools: IPlatformEncryptionTools,
```

Comment:
@inga-affinidi since we going to postpone moving this class to abstract - lets move `platformEncryptionTools ` to the last parameter and if its not provided use default one (default one could be one from browser)

Reply:
Allowing it to be optional and defaulting to browser is too error-prone, we might easily end up accidentally using the one from browser in expo or react-native where it is not supported.
Even if we were to make it optional, I'd say that the default should be the one we have in `common` tests (https://github.com/affinityproject/affinidi-core-sdk/blob/985b097422f40d2f9ba4a4acb2a2ca9e3be17597/common-libs/common/test/factory/CommonNetworkMember.ts) - that is, the default will just not support features that require platform tools.

However, I don't think making it optional is a good idea. There is just too many places where the constructor is called (all these static factories, like `fromSeed`, `signUp` etc). And either they will call `CommonNetworkMember` constructor (and then all these factories have to accept `platformEncryptionTools` as their optional argument, and then every client app that uses react-native SDK will have to provide its platform tools in every call to every such factory, or react-native SDK will have to override all these factories just to pass proper `platformEncryptionTools` to the base class). Or they will call the constructor of an actual non-core SDK, and that means that `CommonNetworkMember` constructor will only be called from non-core SDK constructors, and there is no need in making any of its parameters optional at all.

Reply:
@inga-affinidi In my comment - there I explicitly stayed about this update only in term of backward compatibility , and have it only while we do not updating this class as abstract (just to support exists usage), and this thing should be not optional when we moving to this class to abstract. I was under the assumption that it was clear, so all this your long comment not really required.